lib/prototypes.h

```diff
@@ -299,14 +299,16 @@ struct subid_nss_ops {
 	 *
 	 * @owner - string representing username being queried
 	 * @id_type - subuid or subgid
-	 * @ranges - * a NULL-terminated array of struct subordinate_range,
-	 * or NULL.  The returned array of struct subordinate_range must be
-	 * freed by the caller and it's members, if not NULL.
+	 * @ranges - * an array of struct subordinate_range, or NULL.  The returned
```

Comment:
This is actually a pointer to an array of **pointers** to struct.

This is somewhat inconsistent as compared to `find_subid_owners()`, where out_arg is really a pointer to an array of data.

Do we really need this to be an array of pointers..?

Reply:
Ah... this seems to be deeply embedded into existing code...

Reply:
Hm, I guess we don't.  It could just be a pointer to an array of structs...